code/languages/C++/flow-control/README.md

```diff
@@ -0,0 +1,203 @@
+  }
+
+  cout << "liftoff!\n";
+}
+```
+### Output
+```
+10, 9, 8, 7, 6, 5, 4, 3, 2, 1, liftoff!
+```
+### The do-while loop
+
+A very similar loop is the do-while loop, whose syntax is:
+```
+do statement while (condition);
+```
+
+It behaves like a while-loop, except that condition is evaluated after the execution of statement instead of before, guaranteeing at least one execution of statement, even if condition is never fulfilled. For example, the following example program echoes any text the user introduces until the user enters goodbye:
+
+``` C++
+// echo machine
+#include <iostream>
+#include <string>
+using namespace std;
+
+int main ()
+{
+  string str;
+  do {
+    cout << "Enter text: ";
+    getline (cin,str);
+    cout << "You entered: " << str << '\n';
+  } while (str != "goodbye");
+}
+```
+### Output
+```
+Enter text: hello
+You entered: hello
+Enter text: who's there?
+You entered: who's there?
+Enter text: goodbye
+You entered: goodbye
+```
+## The for loop
+The for loop is designed to iterate a number of times. Its syntax is:
+```
+for (initialization; condition; increase) statement;
+```
+
+Like the while-loop, this loop repeats statement while condition is true. But, in addition, the for loop provides specific locations to contain an initialization and an increase expression, executed before the loop begins the first time, and after each iteration, respectively. Therefore, it is especially useful to use counter variables as condition.
+
+```C++
+// countdown using a for loop
+#include <iostream>
+using namespace std;
+
+int main ()
+{
+  for (int n=10; n>0; n--) {
+    cout << n << ", ";
+  }
+  cout << "liftoff!\n";
+}
+```
+
+### Output
+```
+10, 9, 8, 7, 6, 5, 4, 3, 2, 1, liftoff!
+```
+## Decision
+In this, we test a condition and based on whether the condition is true or false you execute a statement or set of statements. In C++ programming language, we can implement conditional branching using the following statements:
+
+* If….Else Statement
+* switch…case statement
+
+### if-else Statement
+
+When we use the **if-else** statement, you check a condition. If the condition is met (true), a piece of code is executed.  However if the condition is false, the else part of code in the loop body is executed.
+
+The syntax for using the “if” statement is as follows:
+```
+if (condition)
+{
+        Statement 1;
+}
+else
+{
+        Statement 2;
+}
+Statement 3;
+```
+### Sample
+
+```C++
+int main()
+{
+    int age;
+    age = 70;
+    if (age<=60)
+    {
+        cout<<”Correct age”<<endl;
+    }else
+    {
+        cout<<”Incorrect age”<<endl;
+    }
+     
+    return 0;
+}
+```
+### switch statement
+
+The **switch-case** statement tests the value of an expression against a list of conditions. When a match is found, the statements associated with that condition are executed. In the case of a “switch…case” statement, the conditions are constants. Each constant is a “case” (in the body of the “switch….case” statement).
```

Comment:
Not the limitations of what can be compared in a switch. For example, `std::string` cannot be compared in a switch case as easily as one would expect.

Reply:
Done, sir. Please review it.